server/PokerTableObjects/PokerHandRankCalculator.js

```diff
@@ -0,0 +1,425 @@
+            }
+            else if(split[0] === 'J'){
+                number = 11;
+            }
+            else if(split[0] === 'Q'){
+                number = 12;
+            }
+            else if(split[0] === 'K'){
+                number = 13;
+            }
+            else{
+                number = parseInt(split[0]);
+            }
+            let suit = split[1];
+            cards[i+5] = new Card(number, suit);
+        }
+        console.log(JSON.stringify(cards));
+        // we will iterate through the functions until one returns a positive value
+        /*
+        let allFunctions = [
+            this.handStraightFlushValue,
+            this.handQuadsValue,
+            this.handFullHouseValue,
+            this.handFlushValue,
+            this.handStraightValue,
+            this.handThreeOfAKindValue,
+            this.handTwoPairValue,
+            this.handSinglePairValue,
+            this.handHighCardValue
+        ];
+        */
+        // number refers to how many cards to include in the highcard calculation
+        // condition refers to how that function is to return that result, each function
+        // uses condition to its own definition
+        let setUps = [{},{},{},{condition: false}, {condition: true}, {number: 2}, {number: 1}, {number: 3}, {number: 5, condition: true}];
+        
+        let result = -1;
+        /*
+        for(let i = 0; i < allFunctions.length; i++){
+            result = allFunctions[i](cards, setUps[i]); // for some reason this is not working, claims functions are not defined
+            if(result != -1){
+                return result;
+            }
+        }
+        */
+
+
+
+        // NOTE: up until this point I have not found a way to combine all these functions into an array that i can iterate
+        // through, the server crashes (claiming it cannot find my functions and thus I reverted the code back to the messy)
+        // lines of 'if statments' below, do not remove the commented out code above as I might come back to this problem at a
+        // later date and try to fix it.
+
+        result = this.handStraightFlushValue(cards, setUps[0]);
+        if(result !== -1){
+            return result;
+        }
+        result = this.handQuadsValue(cards, setUps[1]);
+        if(result !== -1){
+            return result;
+        }
+        result = this.handFullHouseValue(cards, setUps[2]);
+        if(result !== -1){
+            return result;
+        }
+        result = this.handFlushValue(cards, setUps[3]);
+        if(result !== -1){
+            return result;
+        }
+        result = this.handStraightValue(cards, setUps[4]);
+        if(result !== -1){
+            return result;
+        }
+        result = this.handThreeOfAKindValue(cards, setUps[5]);
+        if(result !== -1){
+            return result;
+        }
+        result = this.handTwoPairValue(cards, setUps[6]);
+        if(result !== -1){
+            return result;
+        }
```

Comment:
All of this logic can be done in a function and returned once

Reply:
i did try that and i kept getting errors in the server, i plan to come back to this if i have the time

Reply:
I'll resolve this comment and we can take a look at it as part of our future refactoring